tokio/src/task/spawn.rs

```diff
@@ -131,7 +131,21 @@ cfg_rt! {
     {
         let spawn_handle = runtime::context::spawn_handle()
         .expect(CONTEXT_MISSING_ERROR);
-        let task = crate::util::trace::task(task, "task");
+        let task = crate::util::trace::task(task, "task", None);
         spawn_handle.spawn(task)
     }
+
+    cfg_trace! {
+        #[cfg_attr(tokio_track_caller, track_caller)]
+        pub fn spawn_named<T>(name: impl Into<std::borrow::Cow<'static, str>>, task: T) -> JoinHandle<T::Output>
```

Comment:
Couple of thoughts.

First, is there a reason to avoid the builder patter? Generally, we try to match the `std` API and specifying a name matches up w/ the [`thread::Builder`](https://doc.rust-lang.org/stable/std/thread/struct.Builder.html) pattern. Also, using a builder could allow us to add `Builder::spawn_blocking` as a separate API.

Then, what was the motivation to use `impl Into<std::borrow::Cow<'static, str>>`? Looking at the code, it **seems** like `&str` should work?

Reply:
To elaborate a bit, we **can** deviate from copying `std` APIs and patterns, but we should have a good reason to do so.